source/app/models/url.rb

```diff
@@ -0,0 +1,12 @@
+class Url < ActiveRecord::Base
+  before_save :sanitize_address
+  before_save :compress_id
+
+  def sanitize_address 
+    self.address.gsub!(/https:\/\/|http:\/\//,'')
+  end
+
+  def compress_id
+    self.id = IdCompressor.compress(Url.count + 1)
```

Comment:
I was honestly trying to keep the number of columns in the db to an absolute minimum, to save on space. But the space savings definitely won't be worth the time complexity of using `.count`. I'll either use a counter_cache or just keep the standard active record id along with the compressed id.